bin/proving-service/Cargo.toml

```diff
@@ -12,60 +12,49 @@ rust-version.workspace = true
 edition.workspace = true
 
 [[bin]]
-name = "miden-tx-prover"
+name = "miden-proving-service"
 path = "src/main.rs"
 required-features = ["std"]
 
-[lib]
-crate-type = ["lib"]
-
 [features]
-async = ["miden-tx/async"]
 default = ["std"]
 std = ["miden-objects/std", "miden-tx/std", "dep:tokio", "dep:tonic-web", "dep:tokio-stream", "dep:axum",  "dep:tracing", "dep:tracing-subscriber", "tonic/transport"]
 testing = ["miden-objects/testing", "miden-lib/testing", "miden-tx/testing"]
 concurrent = ["miden-lib/concurrent", "miden-objects/concurrent", "miden-tx/concurrent", "std"]
```

Comment:
Do we need all these features? Specifically:

- Proving service always runs under `std` - so, not sure we need `default` and `std` features.
- Not sure if `testing` feature is needed any more (what do we use it for?).
- Should we always enable `concurrent`?

Reply:
> Proving service always runs under std - so, not sure we need default and std features.

This isn't needed anymore.

> Not sure if testing feature is needed any more (what do we use it for?).

I think that we should keep it at least until it is removed from the lib, tx and objects crates. We use the feature for the worker, since the proof witness is different if the feature is there or not.

> Should we always enable concurrent?

Maybe the best shot is making the `concurrent` feature enabled by default, but we should keep the ability to don't use it if needed.